designs/sam-debugging/local-sam-debugging.md

```diff
@@ -0,0 +1,308 @@
+
+### SAM Template Resources (API Gateway style Local Invoke)
+
+SAM Template Resources that contain an event of type [Api](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-function-api.html), can be hosted in a web server for local development and iteration. A REST request triggers the Lambda handler as if it were called by API Gateway, and the Lambda function receives an API Gateway based event.
+
+### Standalone Lambda Function Handlers
+
+Lambda Function Handler code can be locally Run or Debugged, even if it does not belong to a SAM Application. The Toolkit produces a temporary SAM Application to contain the handler code. This temporary SAM Application is handled as mentioned [earlier](#sam-template-resource-local). At the end of the debug session, the temporary SAM Application is removed.
+
+In this mode, any SAM Templates that reference a Handler are ignored. This prevents confusion/errors introduced when trying to perform a reverse-lookup between code and SAM Template Resources (examples include incorrectly determining a function's lambda handler string, or situations where more than one resource references the same function).
+
+The Toolkit does not provide support for locally running or debugging standalone Lambda function handlers as API Gateway calls. The code should be referenced from a SAM Template in order to use the API Gateway style debugging features mentioned in the earlier section.
+
+## <a id="debug-config"></a> What can be configured for a Debug session?
+
+The following parameters influence a debug session. These are user-configured, and are referenced by the various [debugging experiences](#debugging-experiences).
+
+| Property                | Description                                          | Used by Standalone Lambda Handler | Used by SAM Template Resources |
+| ----------------------- | ---------------------------------------------------- | --------------------------------- | ------------------------------ |
+| SAM Template            | Path to SAM Template file                            |                                   | x                              |
+| SAM Template Resource   | Name of resource within SAM Template                 |                                   | x                              |
+| SAM Template Parameters | Values to use for SAM Template Parameters            |                                   | x                              |
+| Environment Variables   | Environment Variables exposed to the Lambda Function | x                                 | x                              |
+| Input Event             | Payload passed to the invoked Lambda Function        | x                                 | x                              |
+| Runtime                 | Runtime of Lambda Function to invoke                 | x                                 |                                |
```

Comment:
These things are still 'configurable' they're just inferred from the template - rather than in a separate configuration.

Reply:
Indeed this would be confusing - I've added "obtained from SAM Template" into the "Sam Template Resources" column for these.